server/src/postgis/nearest.rs

```diff
@@ -0,0 +1,361 @@
+//! This module contains functions for routing between nodes.
+use crate::grpc::server::grpc_server::{
+    DistanceTo, NearestNeighborRequest, NodeType as GrpcNodeType,
+};
+
+use crate::postgis::NodeType;
+use uuid::Uuid;
+
+/// Possible errors with path requests
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum NNError {
+    /// No path was found
+    NoPath,
+
+    /// Invalid start node
+    InvalidStartNode,
+
+    /// Invalid end node
+    InvalidEndNode,
+
+    /// Invalid limit
+    InvalidLimit,
+
+    /// Invalid range
+    InvalidRange,
+
+    /// Unsupported path type
+    Unsupported,
+
+    /// Could not get client
+    Client,
+
+    /// Unknown error
+    Unknown,
+}
+
+impl std::fmt::Display for NNError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            NNError::NoPath => write!(f, "No path was found."),
+            NNError::InvalidStartNode => write!(f, "Invalid start node."),
+            NNError::InvalidEndNode => write!(f, "Invalid end node."),
+            NNError::InvalidLimit => write!(f, "Invalid limit."),
+            NNError::InvalidRange => write!(f, "Invalid range."),
+            NNError::Unsupported => write!(f, "Unsupported path type."),
+            NNError::Client => write!(f, "Could not get client."),
+            NNError::Unknown => write!(f, "Unknown error."),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct NNRequest {
+    node_start_id: String,
+    start_type: NodeType,
+    end_type: NodeType,
+    limit: i32,
+    max_range_meters: f64,
+}
+
+/// Sanitize the request inputs
+fn sanitize(request: NearestNeighborRequest) -> Result<NNRequest, NNError> {
```

Comment:
If you had needed the `NNRequest` struct, I think a `TryFrom` implementation for `NNRequest` would have suited us better here.
```
impl TryFrom<NearestNeighborRequest> for NNRequest {
    type Error = NNError;

    fn try_from(request: NearestNeighborRequest) -> Result<Self, Self::Error> {
        ... (your sanitize function content)
    } 
}
```